### ISSUE TITLE
Using Lombok's @Data Annotation Doesn't Work With Error Prone

### ISSUE THREAD

**Issue by nishantc1527** [google/error-prone#1754]
> ATTENTION! Please read and follow:
> - if this is a _question_ about Error Prone, send it to error-prone-discuss@googlegroups.com
> - if this is a _bug_ or _feature request_, fill the form below as best as you can.

### Description of the problem / feature request:

> I have the following code:

```
@Data
@AllArgsConstructor
public static class TableEntry {
    public String lexme;
    public TokenType type;
    public int line;
    public int col;
}
```

The ```@Data``` and ```@AllArgsConstructor``` annotations are part of the ```lombok``` library. This code builds fine. However, using error prone it gives me this stack trace:

```
  @Data
  ^
     Please report this at https://github.com/google/error-prone/issues/new and include the following:
  
     error-prone version: 2.1.3
     Stack Trace:
     java.lang.IndexOutOfBoundsException
        at java.nio.HeapCharBuffer.subSequence(HeapCharBuffer.java:596)
        ... (Many More Exceptions Omitted)
```

### Feature requests: what underlying problem are you trying to solve with this feature?

> A way to suppress this error.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

> Download the [lombok](https://projectlombok.org/) jar file, add it to your classpath, create a class with at least 1 field, then give the class the ```@Data``` annotation.

```
@Data
class MyClass {
    public static int myVar;
}
```

### What version of Error Prone are you using?

> Version 2.1.3

###  Have you found anything relevant by searching the web?

> #1185
> #1214 
The solution they are suggesting is add the ```annotationProcessor``` option with lombok. I did that. Here is the section in my ```build.gradle``` file:

```
compileOnly 'org.projectlombok:lombok:1.18.12'
annotationProcessor 'org.projectlombok:lombok:1.18.12'
```

> Places to look:
> - GitHub issues: https://github.com/google/error-prone/issues
> - email threads on https://groups.google.com/forum/#!forum/error-prone-discuss

**Comment by tbroyer** [comment]
Your ErrorProne version is too old. Many issues with Lombok have been fixed in 2.3.4 and 2.4.0 IIRC.

**Comment by nishantc1527** [comment]
Thanks for that, I updated it.
